File: rust/src/domain/chapter.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Chapter {
    pub id: i64,
    pub manga_id: i64,
    pub url: String,
    pub name: String,
    pub chapter_number: f64,
    pub scanlator: Option<String>,
    pub read: bool,
    pub bookmark: bool,
    pub last_page_read: i32,
    pub date_fetch: Option<i64>,
    pub date_upload: Option<i64>,
    pub source_order: i32,
}

File: lib/src/domain/chapter/model/chapter.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'chapter.freezed.dart';
part 'chapter.g.dart';

@freezed
sealed class Chapter with _$Chapter {
  const factory Chapter({
    required int id,
    required int mangaId,
    required String url,
    required String name,
    @Default(0.0) double chapterNumber,
    String? scanlator,
    @Default(false) bool read,
    @Default(false) bool bookmark,
    @Default(0) int lastPageRead,
    DateTime? dateFetch,
    DateTime? dateUpload,
    @Default(0) int sourceOrder,
  }) = _Chapter;

  factory Chapter.fromJson(Map<String, dynamic> json) =>
      _$ChapterFromJson(json);
}
